### PR TITLE
feat: `avm/res/document-db/database-account` Add the creation of 'Microsoft.DocumentDB/databaseAccounts/sqlRoleDefinitions' and ‘Microsoft.DocumentDB/databaseAccounts/sqlRoleAssignments’ and 'cosmos.properties.documentEndpoint'.

### LINKED ISSUE #1126: [AVM Module Issue]: DocumentDB Database should add the creation of 'Microsoft.DocumentDB/databaseAccounts/sqlRoleDefinitions' and ‘Microsoft.DocumentDB/databaseAccounts/sqlRoleAssignments’ and the output of 'cosmos.properties.documentEndpoint'
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Feature Request

### Module Name

avm/res/document-db/database-account

### (Optional) Module Name if not listed above

avm/res/document-db/database-account

### (Optional) Module Version

_No response_

### Description

Please add the creation of `Microsoft.DocumentDB/databaseAccounts/sqlRoleDefinitions` and `Microsoft.DocumentDB/databaseAccounts/sqlRoleAssignments` the output of `roleDefinition.outputs.id` and `cosmos.properties.documentEndpoint`.
For example (azd): 
https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/database/cosmos/sql/cosmos-sql-role-def.bicep#L4
https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/database/cosmos/sql/cosmos-sql-role-assign.bicep#L7
https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/database/cosmos/sql/cosmos-sql-role-def.bicep#L30
https://github.com/Azure/azure-dev/blob/main/templates/common/infra/bicep/core/database/cosmos/cosmos-account.bicep#L47

CC: @jongio
### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by zedy-wj**
## Description

Fixes #1126 
## Pipeline Reference

| Pipeline |
| -------- |
|[avm.res.document-db.database-account](https://github.com/zedy-wj/bicep-registry-modules/actions/runs/9481159887/job/26165484398)          |

The failure in the pipeline is due to the region hardcoded as `eastasia` in the test files, and in our subscription, there is no permission to create storage in this region. The region in the newly added test file for `sql-role` has been changed to eastus2 and it passed.
## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ x] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ x] I'm sure there are no other open Pull Requests for the same update/change
- [ x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ x] My corresponding pipelines / checks run clean and green without any errors or warnings

@jongio for notification.

### CODE REVIEW

File: avm/res/document-db/database-account/tests/e2e/role/dependencies.bicep [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+@description('Optional. The kind of the App Service Plan.')
+param appServicePlanKind string = ''
+
+@description('Optional. If Linux app service plan , otherwise.')
+param appServicePlanReserved bool = true
+
+@description('Optional. Managed service identity.')
+param managedIdentity bool = !empty(keyVaultName)
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-06-01-preview' = {
```
Comment (bryansan-msft) [inline avm/res/document-db/database-account/tests/e2e/role/dependencies.bicep:33]: This key vault is not used for testing. Can we remove it?

File: avm/res/document-db/database-account/tests/e2e/role/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+param appServicePlanName string
+
+@description('Optional. The kind of the App Service Plan.')
+param appServicePlanKind string = ''
+
+@description('Optional. If Linux app service plan , otherwise.')
+param appServicePlanReserved bool = true
+
+@description('Optional. Managed service identity.')
+param managedIdentity bool = !empty(keyVaultName)
```
Comment (bryansan-msft) [inline avm/res/document-db/database-account/tests/e2e/role/dependencies.bicep:31]: Your test always needs a managed identity. Can we remove this parameter and always emit an identityPrincipalId without a condition?

File: avm/res/document-db/database-account/tests/e2e/role/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,60 @@
+targetScope = 'subscription'
+
+metadata name = 'Deploying with a key vault reference to save secrets'
```
Comment (bryansan-msft) [inline avm/res/document-db/database-account/tests/e2e/role/main.test.bicep:3]: This metadata value is a copy paste of another test. Can we write a meaningful one?

File: avm/res/document-db/database-account/tests/e2e/role/dependencies.bicep [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+@description('Optional. Type of site to deploy.')
+@allowed([
+  'functionapp' // function app windows os
+  'functionapp,linux' // function app linux os
+  'functionapp,workflowapp' // logic app workflow
+  'functionapp,workflowapp,linux' // logic app docker container
+  'app,linux' // linux web app
+  'app' // normal web app
+])
+param appKind string = 'app,linux'
```
Comment (bryansan-msft) [inline avm/res/document-db/database-account/tests/e2e/role/dependencies.bicep:19]: Can we simplify all of these parameters and just hardcode them in the bicep resource directly? your test is not using them and this is just added extra noise

File: avm/res/document-db/database-account/sql-role/sql-role-definitions/main.bicep [hunk 5]
```diff
@@ -0,0 +1,53 @@
+metadata name = 'DocumentDB Database Account SQL Role Definitions.'
+metadata description = 'This module deploys a SQL Role Definision in a CosmosDB Account.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Conditional. The name of the parent Database Account. Required if the template is used in a standalone deployment.')
+param databaseAccountName string
+
+@description('Optional. An array of data actions that are allowed.')
+param dataActions array = [
+  'Microsoft.DocumentDB/databaseAccounts/readMetadata'
```
Comment (bryansan-msft) [inline avm/res/document-db/database-account/sql-role/sql-role-definitions/main.bicep:10]: Can we decide whether these default values are here or in the parent "sql-role/main.bicep" file? These default values are duplicated these two files.

File: avm/res/document-db/database-account/sql-role/main.bicep [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+@description('Conditional. The name of the parent Database Account. Required if the template is used in a standalone deployment.')
+param databaseAccountName string
+
+@description('Required. Name of the SQL Role.')
+param name string
+
+@description('Optional. An array of data actions that are allowed.')
+param dataActions array = [
+  'Microsoft.DocumentDB/databaseAccounts/readMetadata'
```
Comment (bryansan-msft) [inline avm/res/document-db/database-account/sql-role/main.bicep:13]: As i mentioned in my other comment. We need to either have defaults value here or in its children files but not in both as those defaults values are duplicated and generated extra noise and confusion

**Review by ghost — CHANGES_REQUESTED** [review]
Added comments mostly in code simplification regarding unused parameters. Can you please link the workflow run with your new role test in green? the link you posted in the PR description is not opening for me

**Comment by zedy-wj** [comment]
> Added comments mostly in code simplification regarding unused parameters. Can you please link the workflow run with your new role test in green? the link you posted in the PR description is not opening for me

We have updated  the link to the new role test workflow in the description, and also updated the code based on your suggestion, please review. Thanks.

**Review by ghost — APPROVED** [review]
Looks good to me

**Pull request merged**